torch/distributed/fsdp/_common_utils.py

```diff
@@ -390,12 +390,16 @@ def f(module: torch.nn.Module, prefix: str, tree_level: int, *args, **kwargs):
                         submodule_name == "_fsdp_wrapped_module"
                         or submodule_name == "_dmp_wrapped_module"
                     ):
-                        warnings.warn(
-                            "An unexpected prefix is detected. This case "
-                            " should only happen when using DMP with FSDP. "
-                            f"prefix = {prefix}, "
-                            f"submodule_name = {submodule_name}"
-                        )
+                        if (
+                            not torch.distributed._functional_collectives.is_torchdynamo_compiling()
+                        ):
+                            # TODO(voz): Don't graph break on this
```

Comment:
do you have a plan to not graph break on warn?  (not in this PR, what you did here looks good for now)

Reply:
Kind of - its the same as print. TODO is perhaps toooo promise-y? There's a few strategies we can do here.

Reply:
I like to file an issue and then link to the issue, makes it easier to track